Source/SA/DPD/DPD.cs

```diff
@@ -0,0 +1,207 @@
+            specAnSignal.Dpd.Configuration.ConfigureMemoryPolynomial(selectorString, mpConfig.MemoryPolynomialOrder, mpConfig.MemoryPolynomialDepth);
+            specAnSignal.Dpd.Configuration.ConfigureGeneralizedMemoryPolynomialCrossTerms(selectorString, mpConfig.MemoryPolynomialLeadOrder,
+                mpConfig.MemoryPolynomialLagOrder, mpConfig.MemoryPolynomialLeadMemoryDepth, mpConfig.MemoryPolynomialLagMemoryDepth,
+                mpConfig.MemoryPolynomialMaxLead, mpConfig.MemoryPolynomialMaxLag);
+            specAnSignal.Dpd.Configuration.ConfigureIterativeDpdEnabled(selectorString, mpConfig.IterativeDpdEnabled);
+            specAnSignal.Dpd.ApplyDpd.ConfigureMemoryModelCorrectionType(selectorString, mpConfig.MemoryModelCorrectionType);
+        }
+        #endregion
+
+        #region ApplyDPD
+        public static LookupTableResults ApplyLookupTableDPD(RFmxSpecAnMX specAnSignal, NIRfsg rfsgSession, CommonConfiguration commonConfig, LookupTableConfiguration lutConfig, Waveform waveform, string selectorString = "")
+        {
+            LookupTableResults lutResults = new LookupTableResults();
+            IntPtr rfsgPtr;
+            RFmxSpecAnMXDpdApplyDpdIdleDurationPresent idlePresent;
+            idlePresent = waveform.IdleDurationPresent ? RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.True : RFmxSpecAnMXDpdApplyDpdIdleDurationPresent.False;
+            rfsgPtr = rfsgSession.GetInstrumentHandle().DangerousGetHandle();
+            specAnSignal.Initiate("", selectorString);
+            specAnSignal.Dpd.ApplyDpd.ConfigureLookupTableCorrectionType(selectorString, lutConfig.LookupTableCorrectionType);
+            specAnSignal.Dpd.ApplyDpd.ApplyDigitalPredistortion(selectorString, waveform.WaveformData, idlePresent, commonConfig.Timeout, ref lutResults.WaveformWithDpdComplexSingle,
+                out lutResults.PostDPDPAPR, out lutResults.PowerOffset);
+
+            waveform.WaveformData = lutResults.WaveformWithDpdComplexSingle;
+            waveform.SampleRate = 1 / lutResults.WaveformWithDpdComplexSingle.PrecisionTiming.SampleInterval.TotalSeconds;
+            waveform.PAPR_dB = lutResults.PostDPDPAPR;
+            waveform.SignalBandwidth_Hz = 0.8 * waveform.SampleRate;
+
+            SG.DownloadWaveform(rfsgSession, waveform);
+            rfsgSession.RF.PowerLevel = commonConfig.DutAverageInputPower - lutResults.PowerOffset;
+            rfsgSession.Initiate();
+            specAnSignal.Dpd.Results.FetchLookupTable(selectorString, commonConfig.Timeout, ref lutResults.LookupTableInputPowers, ref lutResults.LookupTableComplexGains);
+            return lutResults;
+
+        }
+
+        public static MemoryPolynomialResults ApplyPolynomialDPD(RFmxSpecAnMX specAnSignal, NIRfsg rfsgSession, CommonConfiguration commonConfig, MemoryPolynomialConfiguration mpConfig, Waveform waveform, string selectorString = "")
```

Comment:
See comments on 185 and 194:
```suggestion
        public static MemoryPolynomialResults ApplyPolynomialDPD(RFmxSpecAnMX specAnSignal, NIRfsg rfsgSession, MemoryPolynomialConfiguration mpConfig, 
                Waveform waveform, string selectorString = "", timeout_s = 10)
```